src/eth/EthereumCdpService.js

```diff
@@ -105,9 +105,16 @@ export default class EthereumCdpService extends PrivateService {
     });
   }
 
-  async lockEth(cdpId, amount, unit = ETH) {
-    await this._conversionService().convertEthToWeth(amount, unit);
-    return this.lockWeth(cdpId, amount);
+  lockEth(cdpId, amount, unit = ETH) {
+    const promise = (async () => {
+      await 0;
+      await this._conversionService().convertEthToWeth(amount, unit, {
+        promise,
+        txLabel: 'deposit'
```

Comment:
step 2 variation: if a single method triggers multiple transactions, we have to differentiate between them with some label (this example is still very WIP)

Reply:
On this note, I've been working on formalising the action metadata that the CDP Portal UI consumes as each tx comes through the TransactionManager's `onNewTransaction` event.

So far here's the gist of it: https://gist.github.com/michaelelliot/06441148e89bd9e4688f34f98794d266

If we're using labels, perhaps the action name could be used here instead. (Rather than `txLabel`.)

Reply:
This label doesn't map 1:1 with actions though. "lockEth" causes "weth.deposit", "peth.join", and "tub.lock".